scheduler/src/cook/mesos/task.clj

```diff
@@ -280,12 +280,24 @@
   {"DOCKER" :container-type-docker
    "MESOS" :container-type-mesos})
 
+(defn- assign-port-mappings
+  "Assign port mappings from offer. Port n in the cook job should be mapped to the nth
+   port taken from the offer."
+  [container ports-assigned]
+  (update-in container [:docker :port-mappings]
+             (fn [port-mappings]
+               (map (fn [{:keys [host-port] :as port-mapping}]
+                      (if (contains? ports-assigned host-port)
```

Comment:
Is this `if` necessary? Shouldn't it be an error if the port is not available? 

Please convert it to a `when` if there is no else clause.

Reply:
Technically you can map a port outside the offer, so I wanted to support that case as well. This is the same logic that marathon uses.

There is an else clause, `port-mapping`